skt/__init__.py

```diff
@@ -12,117 +12,75 @@
-        # Add the RPC version checking call/return wrappers
-        return self._return_unwrapper(self._wrap_call(self.rpc, name))
-
-
-# FIXME This doesn't just "parse" patchwork URL. Rename/refactor.
-def parse_patchwork_url(uri):
+def get_patch_mbox(url):
     """
-    Create a Patchwork XML RPC interface and extract the patch ID from a
-    Patchwork patch URL.
+    Retrieve a string representing mbox of the patch.
 
     Args:
-        uri:    The Patchwork patch URL.
+        url: Patchwork URL of the patch to retrieve
 
     Returns:
-        The created Patchwork XML RPC interface and the patch ID string.
+        String representing body of the patch mbox
 
     Raises:
-        str: URL format was not recognized, or an XML RPC
-             compatibility/communication error has occurred and the string
-             describes it.
+        Exception in case the URL is currently unavailable or invalid
     """
-    m = re.match(r"^(.*)/patch/(\d+)/?$", uri)
-    if not m:
-        raise Exception("Can't parse patchwork url: '%s'" % uri)
-
-    baseurl = m.group(1)
-    patchid = m.group(2)
+    # Use os.path for manipulation with URL because urlparse can't deal
+    # with URLs ending both with and without slash.
+    mbox_url = os.path.join(url, 'mbox')
 
-    rpc = xmlrpclib.ServerProxy("%s/xmlrpc/" % baseurl)
     try:
-        ver = rpc.pw_rpc_version()
-        # check for normal patchwork1 xmlrpc version numbers
-        if not (ver == [1, 3, 0] or ver == 1):
-            raise Exception("Unknown xmlrpc version %s", ver)
-
-    except xmlrpclib.Fault as err:
-        if err.faultCode == 1 and \
-           re.search("index out of range", err.faultString):
-            # possible internal RH instance
-            rpc = RpcWrapper(rpc)
-            ver = rpc.pw_rpc_version()
-            if ver < 1010:
-                raise Exception("Unsupported xmlrpc version %s", ver)
-        else:
-            raise Exception("Unknown xmlrpc fault: %s", err.faultString)
+        response = requests.get(mbox_url)
+    except requests.exceptions.RequestException as exc:
+        raise(exc)
+
+    if response.status_code != requests.codes.ok:
+        raise Exception('Failed to retrieve patch from %s, returned %d' %
+                        (url, response.status_code))
+
+    return response.content
+
+
+def get_patch_subject(content):
+    """
+    Retrieve a subject from the mbox string representing a patch.
 
-    return (rpc, patchid)
+    Args:
+        content: String representing patch mbox
+
+    Returns:
+        Subject (name of the patch)
+    """
+    headers = email.parser.Parser().parsestr(content, True)
+    subject = headers['Subject']
+    if not subject:
+        # Emails return None if the header is not found so use a stub subject
+        # instead of it
+        return 'subject'
+
+    try:
+        # decode_header() returns a list of tuples (value, charset)
+        return email.header.decode_header(subject)[0][0]
```

Comment:
Let's go the whole way here. Otherwise we can end with partial subjects with wrong encoding in our reports.

Here's an example of an encoded subject from RFC 2047:

    Subject: =?ISO-8859-1?B?SWYgeW91IGNhbiByZWFkIHRoaXMgeW8=?=
        =?ISO-8859-2?B?dSB1bmRlcnN0YW5kIHRoZSBleGFtcGxlLg==?=

`decode_header()` will return this for it:

    [('If you can read this yo', 'iso-8859-1'), ('u understand the example.', 'iso-8859-2')]

Which we can then feed into a simple loop, using `str.decode()`. Something
like this (not tested):

    decoded_piece_list = []
    for piece in email.header.decode_header(subject):
        decoded_piece_list.append(piece[0].decode(piece[1], "replace"))
    return "".join(decoded_piece_list)

Reply:
Ew. Will fix.